File: process-management-api/src/main/java/com/victorsantos/processmanagementapi/users/usercases/admin/getuser/GetUserUserCaseImpl.java
package com.victorsantos.processmanagementapi.users.usercases.admin.getuser;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import com.victorsantos.processmanagementapi.exceptions.NotFoundException;
import com.victorsantos.processmanagementapi.users.data.UserDataGateway;
import com.victorsantos.processmanagementapi.users.data.UserDataResponse;

@Component
public class GetUserUserCaseImpl implements GetUserUserCase {

  @Autowired
  private UserDataGateway userDataGateway;

  @Override
  public GetUserUserCaseResponse run(String id) {
    return userDataGateway.findById(id).map(this::mapToUserCaseResponse)
        .orElseThrow(() -> new NotFoundException("User not found with id " + id));

  }

  private GetUserUserCaseResponse mapToUserCaseResponse(UserDataResponse userData) {
    return GetUserUserCaseResponse.builder()
        .id(userData.getId())
        .name(userData.getName())
        .email(userData.getEmail())
        .role(userData.getRole())
        .build();
  }

}

File: process-management-web/src/users/admin/usercases/get-user/GetUserUserCaseImpl.ts
import { inject, injectable } from "inversify";
import BFFClient from "../../../../clients/BFFClient";
import { clients } from "../../../../diSymbols";
import GetUsersUserCase from "./GetUserUserCase";
import GetUserUserCaseResponse from "./GetUserUserCaseResponse";

@injectable()
export default class GetUserUserCaseImpl implements GetUsersUserCase {
  constructor(@inject(clients.BFF_CLIENT) private _bffClient: BFFClient) {}

  async run(id: string): Promise<GetUserUserCaseResponse> {
    const { data } = await this._bffClient.get("/users/" + id);
    return new GetUserUserCaseResponse({ ...data });
  }
}
